### PR TITLE
[quantization] Enable groupwise Conv2D in GPTQs

### PR DESCRIPTION
**Pull request by stamalakhov**
This PR makes it possible to quantize `Conv2D` with `groups != 1` using `FPIGPTQ/GPTQ`.

<details> <summary> ./ccex test --include-internal -k test_gptq </summary>

```

UN unit tests with -k test_gptq ...
/mnt/storage/slow_repos/TICO/.venv/lib/python3.10/site-packages/torch/backends/__init__.py:46: UserWarning: Please use the new API settings to control TF32 behavior, such as torch.backends.cudnn.conv.fp32_precision = 'tf32' or torch.backends.cuda.matmul.fp32_precision = 'ieee'. Old settings, e.g, torch.backends.cuda.matmul.allow_tf32 = True, torch.backends.cudnn.allow_tf32 = True, allowTF32CuDNN() and allowTF32CuBLAS() will be deprecated after Pytorch 2.9. Please see https://pytorch.org/docs/main/notes/cuda.html#tensorfloat-32-tf32-on-ampere-and-later-devices (Triggered internally at /pytorch/aten/src/ATen/Context.cpp:80.)
  self.setter(val)
test_groupwise_conv2d (quantization.algorithm.test_gptq.GPTQTest) ... ok
test_model (quantization.algorithm.test_gptq.GPTQTest) ... You are using the default legacy behaviour of the <class 'transformers.models.llama.tokenization_llama_fast.LlamaTokenizerFast'>. This is expected, and simply means that the `legacy` (previous) behavior will be used so nothing changes for you. If you want to use the new behaviour, set `legacy=False`. This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565 - if you loaded a llama tokenizer from a GGUF file you can ignore this message.
ok
test_net (quantization.algorithm.test_gptq.GPTQTest) ... /mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:57: FutureWarning: `torch.export.export_for_training` is deprecated and will be removed in PyTorch 2.10. Please use `torch.export.export` instead, which is functionally equivalent.
  model = torch.export.export_for_training(
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:65: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  model = prepare_pt2e(model, quantizer)
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:81: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  return convert_pt2e(model)
ok
test_net_on_zero_inputs (quantization.algorithm.test_gptq.GPTQTest) ... ok
test_normconv2d (quantization.algorithm.test_gptq.GPTQTest) ... /mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:57: FutureWarning: `torch.export.export_for_training` is deprecated and will be removed in PyTorch 2.10. Please use `torch.export.export` instead, which is functionally equivalent.
  model = torch.export.export_for_training(
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:65: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  model = prepare_pt2e(model, quantizer)
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:81: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  return convert_pt2e(model)
ok
test_normconv2d_on_zero_inputs (quantization.algorithm.test_gptq.GPTQTest) ... ok

----------------------------------------------------------------------
Ran 6 tests in 40.611s

OK
```

</details>

<details> <summary> ./ccex test --include-internal -k test_fpi_gptq </summary>

```
RUN unit tests with -k test_fpi_gptq ...
/mnt/storage/slow_repos/TICO/.venv/lib/python3.10/site-packages/torch/backends/__init__.py:46: UserWarning: Please use the new API settings to control TF32 behavior, such as torch.backends.cudnn.conv.fp32_precision = 'tf32' or torch.backends.cuda.matmul.fp32_precision = 'ieee'. Old settings, e.g, torch.backends.cuda.matmul.allow_tf32 = True, torch.backends.cudnn.allow_tf32 = True, allowTF32CuDNN() and allowTF32CuBLAS() will be deprecated after Pytorch 2.9. Please see https://pytorch.org/docs/main/notes/cuda.html#tensorfloat-32-tf32-on-ampere-and-later-devices (Triggered internally at /pytorch/aten/src/ATen/Context.cpp:80.)
  self.setter(val)
test_groupwise_conv2d (quantization.algorithm.test_fpi_gptq.FPIGPTQTest) ... ok
test_net (quantization.algorithm.test_fpi_gptq.FPIGPTQTest) ... /mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:57: FutureWarning: `torch.export.export_for_training` is deprecated and will be removed in PyTorch 2.10. Please use `torch.export.export` instead, which is functionally equivalent.
  model = torch.export.export_for_training(
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:65: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  model = prepare_pt2e(model, quantizer)
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:81: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  return convert_pt2e(model)
ok
test_net_on_zero_inputs (quantization.algorithm.test_fpi_gptq.FPIGPTQTest) ... ok
test_normconv2d (quantization.algorithm.test_fpi_gptq.FPIGPTQTest) ... /mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:57: FutureWarning: `torch.export.export_for_training` is deprecated and will be removed in PyTorch 2.10. Please use `torch.export.export` instead, which is functionally equivalent.
  model = torch.export.export_for_training(
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:65: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  model = prepare_pt2e(model, quantizer)
/mnt/storage/slow_repos/TICO/tico/quantization/algorithm/pt2e/quantizer.py:81: DeprecationWarning: torch.ao.quantization is deprecated and will be removed in 2.10. 
For migrations of users: 
1. Eager mode quantization (torch.ao.quantization.quantize, torch.ao.quantization.quantize_dynamic), please migrate to use torchao eager mode quantize_ API instead 
2. FX graph mode quantization (torch.ao.quantization.quantize_fx.prepare_fx,torch.ao.quantization.quantize_fx.convert_fx, please migrate to use torchao pt2e quantization API instead (prepare_pt2e, convert_pt2e) 
3. pt2e quantization has been migrated to torchao (https://github.com/pytorch/ao/tree/main/torchao/quantization/pt2e) 
see https://github.com/pytorch/ao/issues/2259 for more details
  return convert_pt2e(model)
ok
test_normconv2d_on_zero_inputs (quantization.algorithm.test_fpi_gptq.FPIGPTQTest) ... ok

----------------------------------------------------------------------
Ran 5 tests in 56.207s

OK
```

</details>

Results of `python test_depthwise_quantization.py` (`bits=8`, `calibration_samples = 100`):
[test_depthwise_quantization.py.txt](https://github.com/user-attachments/files/23601512/test_depthwise_quantization.py.txt)

|model| FPI_speedup (%)|time_FPI_GPTQ (s)|time_GPTQ (s)|acc1_original|acc5_original|acc1_FPI_GPTQ|acc5_FPI_GPTQ|acc1_GPTQ|acc5_GPTQ|
|-|-|-|-|-|-|-|-|-|-|
|mobilenet_v2| **38.02**|3.77|6.08|78.56|93.46|78.64 | 93.48| 78.66 | 93.48|
|mobilenet_v3_small| **39.54**|2.48|4.10|75.01|91.02|74.65 | 91.02 |74.63 | 91.02 |
|mobilenet_v3_large| **46.04**| 3.59|6.65|79.70|94.64|79.62 | 94.68 |79.60| 94.70|
|shufflenet_v2_x1_5| **38.07**| 3.07|4.96|80.38| 94.54|80.20 | 94.48 |80.22 | 94.50 |
|swin_v2_b| **71.86**| 8.50|30.20|88.02 |  98.00|88.06 | 98.02 |88.06 | 98.02|
|swin_v2_t| **71.86**| 3.80 |12.87|86.30 | 97.44|86.26 |97.44 |86.26 | 97.46|

_Speed-up is measured as `speed_up = 100 * (1.0 - time_of_FPI_GPTQ/time_oF_GPTQ)`.

Draft: #415 
Issue: #397

TICO-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

**Comment by dayo09** [comment]
Could you wait a bit for a review? 
@mhs4670go is taking days off, he'll be back next week.

**Comment by stamalakhov** [comment]
> Could you wait a bit for a review? 

@dayo09 
Yes. Sure. Thank you.
